### PR TITLE
test(server): covering authentication with unit and integrations tests 

### PR DESCRIPTION
**Pull request by reynan-dev**
## Ticket

**Title:** ETQ Dev, j’ai une couverture des tests sur toutes les codes authentication.
**Notion Link:** https://www.notion.so/jkergal/ETQ-Dev-j-ai-une-couverture-des-tests-sur-toutes-les-codes-authentication-867b6d1de2fd4ff4ac4c1e34d0480cc4

## Description

- Updated jest.config to include coverage reports,
- Created git ignore to ignore that reports
- Updating prettierignore to ignore that reports
- Creating unit and integrations tests on services and DB configurations.

## Guidelines

🔗 Link your PRs to their related Notion tickets
✍️ Use the commit convention to your pull request title
📃 If your code affects the build, update `README.md`
🚫 Do not merge a PR until all comments are resolved
🗑 Remove your branch after merging

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
 coverage
 
-**/volumes/*
-**/build/*
-**/dist/*
-**/tests/*
+**/volumes/**
+**/build/**
+**/dist/**
+**/tests/**
```
Comment (jkergal) [inline .prettierignore:8]: I think we should also format tests files?

File: server/tests/integrations/Authentication.integration.test.ts [hunk 2]
```diff
@@ -64,4 +68,28 @@ describe('Authentication integration test', () => {
 			});
 		});
 	});
+
+	describe('when signing out', () => {
+		describe('when session token is invalid', () => {
+			it('throw a session not found error', async () => {
```
Comment (jkergal) [inline server/tests/integrations/Authentication.integration.test.ts:74]: "it throws" with a "s" mouhahahahah :D

File: server/tests/integrations/services/BaseServices/create.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	beforeEach(async () => {
+		for (const entity of dataSource.entityMetadatas) {
+			const repository = dataSource.getRepository(entity.name);
+			await repository.query(`TRUNCATE ${entity.tableName} RESTART IDENTITY CASCADE;`);
+		}
+	});
+
+	describe('when try create an user with valid data', () => {
+		describe('when trying to create an element that already exists', () => {
+			it('throw an error not created', async () => {
```
Comment (jkergal) [inline server/tests/integrations/services/BaseServices/create.integration.test.ts:24]: same :)

File: server/tests/integrations/services/BaseServices/find.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { closeDatabase, dataSource, startDatabase } from '../../../../src/db';
+import MemberServices from '../../../../src/services/MemberServices';
```
Comment (jkergal) [inline server/tests/integrations/services/BaseServices/find.integration.test.ts:2]: relative path seems really not pratical, we should setup an alias to use absolute path froms src

**Review by jkergal — APPROVED** [review]
😎

File: server/tests/integrations/services/BaseServices/find.integration.test.ts (hunk 4 above)
Comment (reynan-dev) [reply server/tests/integrations/services/BaseServices/find.integration.test.ts:2]: Implementing absolute path on PR #23

File: server/tests/integrations/Authentication.integration.test.ts (hunk 2 above)
Comment (reynan-dev) [reply server/tests/integrations/Authentication.integration.test.ts:74]: Done

File: server/tests/integrations/services/BaseServices/create.integration.test.ts (hunk 3 above)
Comment (reynan-dev) [reply server/tests/integrations/services/BaseServices/create.integration.test.ts:24]: Done

File: .prettierignore (hunk 1 above)
Comment (reynan-dev) [reply .prettierignore]: removed tests on prettierignore

**Pull request merged**